jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessors.java

```diff
@@ -59,11 +61,22 @@ public static JarProcessor from(
     if (jarType.equals(SPRING_BOOT) && mode.equals(ProcessingMode.packaged)) {
       return new SpringBootPackagedProcessor(jarPath);
     } else if (jarType.equals(SPRING_BOOT) && mode.equals(ProcessingMode.exploded)) {
-      return new SpringBootExplodedProcessor(jarPath, temporaryDirectoryProvider.newDirectory());
+      Path explodedJarRoot = cacheDirectories.getExplodedJarDirectory();
+      // Clear the exploded-jar directory first
+      if (Files.exists(explodedJarRoot)) {
+        MoreFiles.deleteRecursively(explodedJarRoot, RecursiveDeleteOption.ALLOW_INSECURE);
+      }
+
+      return new SpringBootExplodedProcessor(jarPath, explodedJarRoot);
     } else if (jarType.equals(STANDARD) && mode.equals(ProcessingMode.packaged)) {
       return new StandardPackagedProcessor(jarPath);
     } else {
-      return new StandardExplodedProcessor(jarPath, temporaryDirectoryProvider.newDirectory());
+      Path explodedJarRoot = cacheDirectories.getExplodedJarDirectory();
+      // Clear the exploded-jar directory first
+      if (Files.exists(explodedJarRoot)) {
+        MoreFiles.deleteRecursively(explodedJarRoot, RecursiveDeleteOption.ALLOW_INSECURE);
+      }
```

Comment:
I wasn't sure who should empty the directory, e.g., if `StandardExplodedProcessor` should clear it instead of `JarProcessors`. I think either makes sense, but if we want the caller to clear it, we should check the precondition inside `StandardExplodedProcessor` to verify that the directory does not exist or is empty.

Otherwise, LGTM, but I'll take a closer look at Monday again.

Reply:
Thanks for pointing this out! I moved the code block to `SpringBootExplodedProcessor` and `StandardExplodedProcessor`  to keep the logic in `JarProcessors` simple.